src/main/java/com/keniu/services/BookServiceImpl.java

```diff
@@ -28,8 +29,15 @@ public List<BookDto> findAll() {
     }
 
     @Override
-    public BookDto getBookById(Long id) {
+    public BookDto getById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id)
             .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id)));
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.toModel(createBookRequestDto);
+        book.setId(id);
```

Comment:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> ......
        bookMapper.updateBookFromDto(bookDto, book);
```

Reply:
I don’t understand you. What's BookDto in your case? Where did you get it from? What's wrong with my approach? What do you suggest to fix or improve? Please clarify

Reply:
You must ensure that the book the user wants to update exists. To do this, find book by ID and update the fields according to BookDto( add the separate method to mapper)

Reply:
I understood what she's saying. However, the thing is, I don't need to check if the entity exists. Spring Data JPA combines both saving and updating an object in the save() method. If the object already exists, it will be updated; if not, a new one will be created. No problem. The task specifies that we should receive the complete object from the outside, which means we are inserting all fields into the database, and we already have all the necessary data for the query. There's no need to go back to the database and fetch what we already have. No sense at all. If we were only receiving a single field to update, then we would use the @Modifying annotation and the @Query("UPDATE Book ...."), and that would be a different story.

Reply:
The solution to update an entity is: 
find book by id in DB
use mapper to update it `bookMapper.updateBookFromDto(createBookRequestDto, book);`
save book and return dto

I know that validation is not part of this task, I texted you my arguments by covered all sides of this topic

Reply:
Once again, I don't need any data from the database to update entity within the scope of this task

Reply:
As mentors, we give advice based on our experience and the latest practices that are relevant in the industry at the moment. However, you have the right to disagree with us. We reserve this right for you.